### PR TITLE
usb: port and bus param parsing

### PR DESCRIPTION
**Pull request by bgartzi**
Several usb device configuration parameters were not being read during device generation in virttest. That implied introducing changes that also affected the name of the keys used to configure those device parameters (please refer to https://github.com/avocado-framework/avocado-vt/pull/3361)

This changes updates the cfg files so the parameters will be taken into account once the mentioned PR gets merged.
It also reverts 75f26e28ae21e4e49b6dfbd1858994cf28362ff1 since the added configuration in that commit will be implied in the base configuration if the already mentioned pr gets merged.

ID: 2059254

### CODE REVIEW

**Comment by bgartzi** [comment]
This one could be tricky and could open further discussion in this, and in the mentioned avocado-vt PR. @YongxueHong, what do you think about it? Should I also mention other maintainers who could be also interested?

**Comment by luckyh** [comment]
@yanan-fu @ybduan would you mind to help review this PR? thanks in advance!

**Comment by yanan-fu** [comment]
@bgartzi  Could you please check the commit `Merge branch 'autotest:master' into bz-2059254`,  seems it is a dirty commit, not related with this patch. Thanks!

**Comment by bgartzi** [comment]
Sure @yanan-fu, and sorry, I didn't notice that before.

**Comment by ybduan** [comment]
I think the following part should be updated, too.
https://github.com/autotest/tp-qemu/blob/02c38ca869ca3c98f548141c0a953bac1483f356/qemu/tests/cfg/usb_device_check.cfg#L99-L110

**Comment by bgartzi** [comment]
Thank you for the catch @ybduan, I'm updating the patch right now.

**Comment by bgartzi** [comment]
I'm also working on the changes needed to adopt the suggestion made by @yanan-fu in https://github.com/avocado-framework/avocado-vt/pull/3361#issuecomment-1058888021 related to the `usb_type` parameter.

**Comment by ybduan** [comment]
This line need to update together:
https://github.com/autotest/tp-qemu/blob/a58c351767727f0084a60cd596f0b28cb9cf234c/qemu/tests/usb_common.py#L56
```suggestion
        usb_dev_id = "usb-%s" % key[12:]
```
Since we updated "usb_type_d0" to "usbdev_type_d0", for example.

**Comment by ybduan** [comment]
I think it's better to update all "usb_type" to "usbdev_type" in function 'parse_usb_topology':

```suggestion
def parse_usb_topology(params):
    """
    Parse the usb devices topology to the params.

    :param params: Dictionary with the test parameters.
    :return: A list of dictionary ({usbdev_type_d0: usbdev_type}) of specified
             usb topology.
    """
    params["usb_devices"] = ""
    # usb topology
    usb_topology = json.loads(params["usb_topology"],
                              object_pairs_hook=OrderedDict)
    parsed_devs = []
    for key, value in usb_topology.items():
        for i in range(value):
            params["usb_devices"] += " d%s" % len(parsed_devs)
            usbdev_type = '{"usbdev_type_d%s": "%s"}' % (len(parsed_devs), key)
            params.update(json.loads(usbdev_type))
            parsed_devs.append(json.loads(usbdev_type))
    return parsed_devs
```

**Comment by ybduan** [comment]
Do you know who are in charge of the changed files except that belong to usb feature? I think they also need to do tests on the changes. Thanks! @yanan-fu

File: qemu/tests/cfg/block_with_share_rw.cfg [hunk 1]
```diff
@@ -22,7 +22,7 @@
                     aarch64:
                         no Host_RHEL
                     usbs = " usbtest"
-                    usb_bus = "usbtest.0"
+                    usbdev_bus = "usbtest.0"
```
Comment (yanan-fu) [inline qemu/tests/cfg/block_with_share_rw.cfg:25]: @qingwangrh  Could you please review this case ? Thanks!

File: qemu/tests/cfg/interrupt_check.cfg [hunk 2]
```diff
@@ -19,7 +19,7 @@
                 - usb_device:
                     irq_pattern = "xhci_hcd"
                     usbs = usb_xhci
-                    usb_bus = "usb_xhci.0"
+                    usbdev_bus = "usb_xhci.0"
```
Comment (yanan-fu) [inline qemu/tests/cfg/interrupt_check.cfg:22]: @PaulYuuu  Seems this test was added by you, can you help review ? Thanks!

File: qemu/tests/cfg/multi_disk_random_hotplug.cfg [hunk 3]
```diff
@@ -55,7 +55,6 @@
                 stg_params = "fmt:virtio,virtio_scsi,usb2"
             usbs += " ehci"
             usb_type_ehci = usb-ehci
-            bus_tablet1 = usb1.0
```
Comment (yanan-fu) [inline qemu/tests/cfg/multi_disk_random_hotplug.cfg:58]: @qingwangrh @YongxueHong Could you help review or forward to the right person if i am wrong ? Thanks!

File: qemu/tests/cfg/pvpanic_basic.cfg [hunk 4]
```diff
@@ -47,7 +47,7 @@
             only Windows
             crash_method = usb_keyboard
             usb_devices += " kb1"
-            usb_type_kb1 = "usb-kbd"
+            usbdev_type_kb1 = "usb-kbd"
```
Comment (yanan-fu) [inline qemu/tests/cfg/pvpanic_basic.cfg:50]: @leidwang Could you help review this case ? Thanks!

File: qemu/tests/cfg/slof_user_interface.cfg [hunk 5]
```diff
@@ -11,7 +11,7 @@
     boot_order = cdn
     boot_strict = off
     usb_devices += ' kbd1'
-    usb_type_kbd1 = 'usb-kbd'
+    usbdev_type_kbd1 = 'usb-kbd'
```
Comment (yanan-fu) [inline qemu/tests/cfg/slof_user_interface.cfg:14]: @YongxueHong  Could you help review the 4 slof related cases ? Thanks!

Comment (YongxueHong) [reply qemu/tests/cfg/slof_user_interface.cfg:14]: > @YongxueHong Could you help review the 4 slof related cases ? Thanks!

Hi @MiriamDeng 
Would you like to help review those? Thanks.

File: qemu/tests/cfg/interrupt_check.cfg (hunk 2 above)
Comment (PaulYuuu) [reply qemu/tests/cfg/interrupt_check.cfg:22]: It's good to me based on the avocado-vt patch.

**Comment by bgartzi** [comment]
Thank you @ybduan for the catches, they've been already applied!
Thank you @yanan-fu for mentioning related people to review this.

**Comment by qingwangrh** [comment]
Passed
python ConfigTest.py --testcase=block_with_share_rw..with_xhci,multi_disk_random_hotplug..all_types --guestname=RHEL.8.4.0 --clone=no

LGTM

File: qemu/tests/cfg/pvpanic_basic.cfg (hunk 4 above)
Comment (leidwang) [reply qemu/tests/cfg/pvpanic_basic.cfg:50]: LGTM.

JOB LOG    : /root/avocado/job-results/job-2022-03-11T04.53-17bc0d8/job.log
 (1/2) Host_RHEL.m.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.q35: STARTED
| (1/2) Host_RHEL.m.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.q35: PASS (1164.08 s)
 (2/2) Host_RHEL.m.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win.x86_64.io-github-autotest-qemu.pvpanic_basic.usb_keyboard.q35: STARTED
\ (2/2) Host_RHEL.m.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win.x86_64.io-github-autotest-qemu.pvpanic_basic.usb_keyboard.q35: PASS (206.90 s)

**Comment by yanan-fu** [comment]
> Passed python ConfigTest.py --testcase=block_with_share_rw..with_xhci,multi_disk_random_hotplug..all_types --guestname=RHEL.8.4.0 --clone=no
> 
> LGTM

Thanks @qingwangrh , how about `multi_disk_random_hotplug` ?

**Comment by qingwangrh** [comment]
> > Passed python ConfigTest.py --testcase=block_with_share_rw..with_xhci,multi_disk_random_hotplug..all_types --guestname=RHEL.8.4.0 --clone=no
> > LGTM
> 
> Thanks @qingwangrh , how about `multi_disk_random_hotplug` ?

Above tests already cover

**Comment by yanan-fu** [comment]
> > > Passed python ConfigTest.py --testcase=block_with_share_rw..with_xhci,multi_disk_random_hotplug..all_types --guestname=RHEL.8.4.0 --clone=no
> > > LGTM
> > 
> > 
> > Thanks @qingwangrh , how about `multi_disk_random_hotplug` ?
> 
> Above tests already cover

yeah, sorry for missing it,  resolve the conversation now, thanks!

File: qemu/tests/cfg/slof_user_interface.cfg (hunk 5 above)
Comment (MiriamDeng) [reply qemu/tests/cfg/slof_user_interface.cfg:14]: I'm afraid that there's something wrong, ran the related jobs on ppc and i got
(status: 1,    output: "qemu-kvm: -device qemu-xhci,id=usb-kbd1,bus=usb1.0,port=2: Device 'qemu-xhci' can't go on usb-bus bus")

**Comment by MiriamDeng** [comment]
Ran the related jobs on ppc and i got the following error message.
--testcase=slof_open_bios,slof_order,slof_user_interface
(status: 1,    output: "qemu-kvm: -device qemu-xhci,id=usb-kbd1,bus=usb1.0,port=2: Device 'qemu-xhci' can't go on usb-bus bus")

**Comment by bgartzi** [comment]
> Ran the related jobs on ppc and i got the following error message after applying this patch.

Sorry :( I'm trying to provision a system on which to debug the issue. I'm getting back to all of you when reaching some meaningful conclusion/fix.

Thanks for all the help everyone!

**Comment by MiriamDeng** [comment]
> Ran the related jobs on ppc and i got the following error message after applying this patch. Could you please re-check ppc part ? Thanks a lot --testcase=slof_open_bios,slof_order,slof_user_interface (status: 1, output: "qemu-kvm: -device qemu-xhci,id=usb-kbd1,bus=usb1.0,port=2: Device 'qemu-xhci' can't go on usb-bus bus")
Unfortunately, still hit the same issue while applying patch https://github.com/avocado-framework/avocado-vt/pull/3361 and this one.

**Comment by bgartzi** [comment]
Hi @MiriamDeng,

I ran some related tests last week, and I was able to hit the issue, and also to find the reason the issue could be happening, thus fixing it.

IIUC you are installing the environment inside a `--venv`, but not using the `--develop` option. After installing avocado-vt then you're applying the patches independently. Correct me if I'm wrong, please. In that case applying the patches over avocado-vt under `<workspace>/avocado-vt` doesn't really mean a full installation of the patch. In fact, the interpreter will look under `<workspace>/lib/python3.x/site-packages/avocado-framework-plugin-vt.egg` or something similar, which will contain the configuration and source files that had been installed when the environment was bootstrapped. Although the patch was applied over avocado-vt in `<workspace>/avocado-vt`, it doesn't mean the files under the previously mentioned directory were modified. I'd say there are two main ways to ensure that the applied patch is taken into account by the interpreter:
* Installing the environment and packages in `--develop` mode
* Installing the environment and packages in whichever mode you want, then reinstalling avocado-vt once the patch has been applied by calling `python setup.py install` under `<workspace>/avocado-vt`.

It worked for me in both the ways I mentioned. The issue wouldn't be hit if the patch is merged to the upstream since the patch would be included and taken into account from the moment it's been installed on.

Now I will describe the ways I checked which could be the issue happening to you, and tested that the patch was working for me. 

### First trial: `--venv` installation
1- I installed the environment inside a virtual environment using the `--venv` argument
2- I applied the patches under the provider repo and under the downloaded avocado-vt repo.
3- I ran the tests.
4- I hit the same issue as you.
5- I reinstalled the avocado-vt package again (with the patch applied) by running the command I mentioned (`python setup.py install` under `<workspace>/avocado-vt`).
6- I ran the tests.
7- They worked fine.

Personal conclusion: the issue is related since the repo files are not being taken into account by the interpreter. Once installed (files are copied inside the packages directory of the virtual environment) it worked.

### Second trial: `--venv --develop` installation
1- I installed the environment inside a virtual environment using the `--venv` argument in develop mode, i.e. using the `--develop` argument.
2- I applied the patches under the provider repo and under the downloaded avocado-vt repo.
3- I ran the tests.
4- They worked without any further effort.

Personal conclusion: When the environment is bootstrapped using `--venv --develop`, the source files are not copied inside the `<..>/lib/python3.x/site-packages/<package>` directory. Instead a link is created to the place the repo is located, i.e. to `<workspace>/avocado-vt`. That way the changes made to those files are taken into account immediately since the interpreter will be reading from them (due to the link). Thus, not needing to reinstall the packages.

To sum up, IMHO the issue is being hit for the way the environment is being installed and the way the patch from avocado-vt is being applied. Merging patches shouldn't be generating a regression in any of the affected repositories. I'd appreciate it if you could check if any of the previously mentioned ways also work with you.

Thank you!

**Comment by MiriamDeng** [comment]
...
> To sum up, IMHO the issue is being hit for the way the environment is being installed and the way the patch from avocado-vt is being applied. Merging patches shouldn't be generating a regression in any of the affected repositories. I'd appreciate it if you could check if any of the previously mentioned ways also work with you.
> 
> Thank you!
Hi bgartzi, thanks for your comments
Also call for @YongxueHong

**Comment by ybduan** [comment]
LGTM.
Acked-by: Yanbin Duan <yduan@redhat.com>

**Comment by MiriamDeng** [comment]
LGTM.  Acked-by: MiriamDeng mdeng@redhat.com 
(1/3) Host_RHEL.m8.u7.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.slof_user_interface.send_f12: STARTED
(1/3) Host_RHEL.m8.u7.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.slof_user_interface.send_f12: PASS (117.90 s)
(2/3) Host_RHEL.m8.u7.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.slof_user_interface.send_boot: STARTED
(2/3) Host_RHEL.m8.u7.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.slof_user_interface.send_boot: PASS (85.26 s)
(3/3) Host_RHEL.m8.u7.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.slof_user_interface.send_reset_all: STARTED
(3/3) Host_RHEL.m8.u7.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.slof_user_interface.send_reset_all: PASS (100.85 s)
